### PR TITLE
Updated to work with latest langchain API

### PR DESCRIPTION
**Pull request by HemuManju**
This PR update the langchain requirement to the latest version (>=0.2.6). The only change required is the requirement of langchain-community package.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -11,15 +11,14 @@
     history = history_file.read()
 
 requirements = [
-    "gptcache<=0.1.35",
-    "langchain<=0.0.226",
+    "langchain>=0.2.6",
```
Comment (jeremymanning) [inline setup.py:14]: should this be strictly greater than?

**Review by jeremymanning — COMMENTED** [review]
this is good overall, but i also get this warning when i submit a query:
Screenshot 2024-06-29 at 8 01 14 PM

I'm using [this notebook](https://colab.research.google.com/drive/1RUlyDZFr6FYaRHrqHgpnooiIzIoNJ7Un?usp=sharing) to test.

i've added a few other minor suggestions too

File: setup.py (hunk 1 above)
Comment (jeremymanning) [reply setup.py:14]: or...do we even need to pin a version?

File: chatify/prompts/tester.yaml [hunk 2]
```diff
@@ -2,36 +2,36 @@ test my understanding with some open-ended questions:
...
-    
+
     ---
     {text}
     ---
 
-    I'd like the correct answer to be either "[A]", "[B]", "[C]", or "[D]". Can you make up a multiple choice question for me so that I can make sure I really understant the most important concepts?  Remember that I can't respond to you, so just ask me to "think about" which choice is correct or something else like that (i.e., without explicitly responding to you). Put two line breaks ("<br>") between each choice so that it appears correctly on my screen. In other words, there should be two line breaks between each of [B], [C], and [D].
+    I'd like the correct answer to be either "[A]", "[B]", "[C]", or "[D]". Can you make up a multiple choice question for me so that I can make sure I really understand the most important concepts?  Remember that I can't respond to you, so just ask me to "think about" which choice is correct or something else like that (i.e., without explicitly responding to you). Put two line breaks ("<br>") between each choice so that it appears correctly on my screen. In other words, there should be two line breaks between each of [B], [C], and [D].
```
[the remark this answers is not in the document]
Comment (jeremymanning) [reply chatify/prompts/tester.yaml:33]: actually: we'll need to update chatify-server for this change to actually take effect.  but still good to make this change.

File: setup.py (hunk 1 above)
Comment (HemuManju) [reply setup.py:14]: I agree with you, no need to pin the version.

File: chatify/chains.py [hunk 3]
```diff
@@ -77,7 +76,8 @@ def __init__(self, config):
         self.llm_models_factory = ModelsFactory()
 
         self.cache = config["cache_config"]["cache"]
```
[the remark this answers is not in the document]
Comment (HemuManju) [reply chatify/chains.py:78]: Yes, that would work!

**Pull request merged**